app/policies/user_policy.rb

```diff
@@ -3,23 +3,24 @@ def index?
     true
   end
 
+  def show?
+    @record.id == @user.id || super
+  end
+
   def create?
-    false
+    @user.roles.none?
   end
 
   def edit?
     false
   end
 
   def update?
-    return true if @record.id == @user.id
-    return true if @user.role?("admin")
-    @user.role?("manager") && !(@record.role?("admin") || @record.role?("manager"))
+    @user.role?("admin") || @record.id == @user.id
   end
 
   def destroy?
-    return true if @user.role?("admin")
-    @record == @user
+    @user.role?("admin")
```

Comment:
we should best not allow this, as users will still be referenced throughout the app (i.e. `created_by_id`, `updated_by_id`) to deleted users. 

instead admins can just demote users (strip their role associations) when they no longer need access

Reply:
@tmfrnz OK makes sense - I'll update the specs to ensure nobody can delete anybody and the demotion can happen through the roles functionality instead.